src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSService.java

```diff
@@ -289,6 +290,24 @@ TaskDefinition registerTemplate(final ECSCloud cloud, final ECSTaskTemplate temp
         }
     }
 
+    void removeTemplate(final ECSCloud cloud, final ECSTaskTemplate template) {
+        AmazonECS client = getAmazonECSClient();
+
+        String familyName = fullQualifiedTemplateName(cloud, template);
+
+        int revision = findTaskDefinition(familyName).getRevision();
+
+        try {
+            client.deregisterTaskDefinition(
```

Comment:
This introduces the need for a new AWS IAM permission.
It should be documented here https://wiki.jenkins.io/display/JENKINS/Amazon+EC2+Container+Service+Plugin (under `IAM Role`)

Reply:
agree, actually cleaning up the templates was completely missing before